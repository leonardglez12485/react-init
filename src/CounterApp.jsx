import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {
    const [counter, setCounter] = useState( value )
    const handleAdd = ()=>{
       setCounter((c)=> c + 1);
    }

    const handleSustract = ()=>{
       setCounter((c)=> c - 1);
    }

    const handleReset = ()=>{
       setCounter(value);
    }

  return (
    <>
       <h1>CounterApp</h1>
       <h2> { counter } </h2>
       <button onClick={handleAdd}>
       +
       </button>
       <button onClick={handleSustract}>
       -
       </button>
       <button onClick={handleReset}>
       Reset
       </button>
    </>
    
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};



