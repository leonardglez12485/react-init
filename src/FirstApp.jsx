
import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle}) => {
    if(!title) throw new Error('Title not exist')
  return (
    <>
     <h2>{ title}</h2>
     <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
 title: 'No title',
  subtitle: 'No subtitle',
}
