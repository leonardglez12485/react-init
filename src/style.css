html, body{
    color: white;
    background-color: #21232A;
    font-family: Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    background-color: #2C2F33;
    color: white;
    border: none;
    padding: 10px 20px;
    font-size: 1.3rem;
    cursor: pointer;
    margin-right: 5px;
}