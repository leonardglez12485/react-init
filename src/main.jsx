import React from 'react';
import ReactDOM from 'react-dom/client';
//import { HelloWorldApp } from './HelloWorldApp';
//import { FirstApp } from './FirstApp';
import './style.css'
import { CounterApp } from './CounterApp';




ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
      {/*<HelloWorldApp/>*/}
     <CounterApp value={25}/>
{  /*  <FirstApp title=" Hola Pepe" subtitle="Hi" name="Leonard"/> */}    </React.StrictMode>
)
