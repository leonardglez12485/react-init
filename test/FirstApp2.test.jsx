import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('<FirstApp Component/>', ()=>{
   
    const title = 'Hola pepe';
    const subtitle= 'Hi';
    const name = 'Leonard';
   
    test('should be match whit snapshot',()=>{
        const {container} =render(<FirstApp title={title} subtitle={subtitle} name={name}/>)
        expect(container).toMatchSnapshot();
    })

    test('should be display mesage: Hola pepe', ()=>{
        render(<FirstApp title={title} subtitle={subtitle} name={name}/>)
        expect(screen.getByText(title)).toBeTruthy;
    })

    test('must display the title in an h2', ()=>{
        render(<FirstApp title={title} subtitle={subtitle} name={name}/>)
        expect(screen.getByRole('heading',{level:2}).innerHTML).toBeTruthy;
        expect(screen.getByRole('heading',{level:2}).innerHTML).toContain(title);
    })
        
    // test('must display the title in an h2', ()=>{
    //     const title = 'Hola pepe';
    //     const subtitle= 'Hi';
    //     const name = 'Leonard';
    //     const {container, getByText, getByTestId} =render(<FirstApp title={title} subtitle={subtitle} name={name}/>)
    //     expect(getByText(title)).toBeInTheDocument;
    //     expect(getByText(title)).toBeTruthy();
        
    //     // const h2 = container.querySelector('h2');
    //     // expect(h2.innerHTML).toContain(title);
    //     expect(getByTestId('test-title')).toBeInTheDocument;
    //     expect(getByTestId('test-title').innerHTML).toContain(title);
    // })
    
    // test('must display the subtitle sent in props', ()=>{
    //     const title = 'Hola pepe';
    //     const subtitle= 'Hi';
    //     const name = 'Leonard';
    //     const {getByText} =render(<FirstApp title={title} subtitle={subtitle} name={name}/>)
     
    //     expect(getByText(subtitle)).toBeTruthy();
        
    //     // const h2 = container.querySelector('h2');
    //     // expect(h2.innerHTML).toContain(title);
    //     // expect(getByTestId('test-title')).toBeInTheDocument;
    //     // expect(getByTestId('test-title').innerHTML).toContain(title);
    // })
})