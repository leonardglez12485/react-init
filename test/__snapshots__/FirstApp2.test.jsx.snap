// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstApp Component/> should be match whit snapshot 1`] = `
<div>
  <h2
    data-testid="test-title"
  >
    Hola pepe
  </h2>
  <p>
    Hi
  </p>
  <p>
    Leonard
  </p>
</div>
`;
